### PR TITLE
Propagate `unit_electrode_indices` to `SortingInterface`

### PR DESCRIPTION
**Pull request by h-mayorquin**
This will allow the user to pass a mapping between units and the electrodes on the electrode table. Eventually I will use this to add a more robust solution like here:

https://github.com/catalystneuro/neuroconv/issues/1112#issuecomment-2422864725

### CODE REVIEW

File: src/neuroconv/nwbconverter.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def validate_conversion_options(self, conversion_options: dict[str, dict]):
 
     def create_nwbfile(self, metadata: Optional[dict] = None, conversion_options: Optional[dict] = None) -> NWBFile:
         """
-        Create and return an in-memory pynwb.NWBFile object with this interface's data added to it.
+        Create and return an in-memory pynwb.NWBFile object with the conversion data added to it.
```
Comment (h-mayorquin) [inline src/neuroconv/nwbconverter.py:147]:
```suggestion
        Create and return an in-memory pynwb.NWBFile object with this interface's data added to it.
```

File: src/neuroconv/datainterfaces/ecephys/basesortingextractorinterface.py [hunk 2]
```diff
@@ -312,9 +313,15 @@ def add_to_nwbfile(
         units_name : str, default: 'units'
             The name of the units table. If write_as=='units', then units_name must also be 'units'.
         units_description : str, default: 'Autogenerated by neuroconv.'
+        unit_electrode_indices : list of lists of int, optional
+            A list of lists of integers indicating the indices of the electrodes that each unit is associated with.
```
Comment (pauladkisson) [inline src/neuroconv/datainterfaces/ecephys/basesortingextractorinterface.py:317]: This corresponds to the sparsity in spikeinterface right?

**Review by pauladkisson — COMMENTED** [review]
This looks good. 

What do you think about adding some kind of how-to explaining what to do if you have 2 probes and run kilosort like in #1112? 

I think it could be an effective way to communicate to users about this new feature.

**Comment by pauladkisson** [comment]
> What do you think about adding some kind of how-to explaining what to do if you have 2 probes and run kilosort like in #1112?

Actually, if this is supposed to be used in `SortedRecordingConverter` specifically, maybe just a conversion gallery entry for that one would be sufficient...

**Comment by h-mayorquin** [comment]
> > What do you think about adding some kind of how-to explaining what to do if you have 2 probes and run kilosort like in #1112?
> 
> Actually, if this is supposed to be used in `SortedRecordingConverter` specifically, maybe just a conversion gallery entry for that one would be sufficient...

Yeah, I would be adding a tutorial there.

**Review by pauladkisson — APPROVED** [review]

**Pull request merged**